Swift/ReplicatorConfiguration.swift

```diff
@@ -58,7 +58,15 @@ public struct ReplicatorConfiguration {
     
     /// The local database to replicate with the replication target.
     @available(*, deprecated, message: "Use config.collections instead.")
-    public let database: Database
+    public var database: Database {
+        guard let db = _db else {
+            NSException(name: .internalInconsistencyException,
+                        reason: "Attempt to access database property but no collections added",
+                        userInfo: nil).raise()
+            fatalError()
```

Comment:
I think now we are mixing b/w using fatalError() and NSException() in Swift code. I think we should decide which one to use. If you are relying on NSException in test code, you can continue to use NSException but please change the rest of the code in Swift to all use NSException for consistency. You can decide here.

Create function to throw exception so that you don't need to call fatalError().
```
throwException(e: NSException) -> Never() {
    e.raise();
}
```

Reply:
```
throwException(e: NSException) -> Never() {
    e.raise();
}
```

Above function return type is `Void` and not `Never`. For that either we should add `fatalError()` or `abort()`

Reply:
I am only worried about the testing when fatalError or abort is used, instead of the NSException. I am more inclined to use fatalError, since that's more swift style.

Reply:
You can decide on this.

Reply:
will use the fatalError